pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -198,6 +218,25 @@ func getMonitoringConsoleStatefulSet(client splcommon.ControllerClient, cr splco
 		},
 	}
 
+	// prepare and append persistent volume claims if storage is not ephemeral
+	var err error
+	statefulSet.Spec.VolumeClaimTemplates, err = getSplunkVolumeClaims(cr, spec, labels)
```

Comment:
I think we better avoid any spec-specific dependencies completely. Idea is, to generate the MC STS, and Pod templates that should be unique irrespective of the CR type, and their configurations.
For the Image parameter, I put a story CSPL-555. But, probably, we can use that to remove the dependency overall(?).

Reply:
Sure, We can make CSPL-555 a generic story to remove any spec dependencies

Reply:
Anything we need from this PR? @sgontla